htdocs/core/triggers/dolibarrtriggers.class.php

```diff
@@ -30,61 +31,76 @@ abstract class DolibarrTriggers
 	 * Name of the trigger
 	 * @var mixed|string
 	 */
-	public $name = '';
+	public $name;
 
 	/**
 	 * Description of the trigger
 	 * @var string
 	 */
-	public $description = '';
+	public $description;
 
 	/**
 	 * Version of the trigger
 	 * @var string
 	 */
-	public $version = self::VERSION_DEVELOPMENT;
+	public $version;
 
 	/**
 	 * Image of the trigger
 	 * @var string
 	 */
-	public $picto = 'technic';
+	public $picto;
 
 	/**
 	 * Category of the trigger
 	 * @var string
 	 */
-	public $family = '';
+	public $family;
 
 	/**
 	 * Error reported by the trigger
 	 * @var string
 	 * @deprecated Use $this->errors
 	 * @see $errors
 	 */
-	public $error = '';
+	public $error;
 
 	/**
 	 * Errors reported by the trigger
 	 * @var array
 	 */
-	public $errors = array();
+	public $errors;
 
 	/**
 	 * @var string module is in development
+	 * @deprecated Use self::VERSIONS
+	 * @see self::VERSIONS
 	 */
 	const VERSION_DEVELOPMENT = 'development';
 
 	/**
 	 * @var string module is experimental
+	 * @deprecated Use self::VERSIONS
+	 * @see self::VERSIONS
 	 */
 	const VERSION_EXPERIMENTAL = 'experimental';
 
 	/**
 	 * @var string module is dolibarr ready
+	 * @deprecated Use self::VERSIONS
+	 * @see self::VERSIONS
 	 */
 	const VERSION_DOLIBARR = 'dolibarr';
 
+	/**
+	 * @var array dictionary of possible module states
+	 */
+	const VERSIONS = [
```

Comment:
What is the benefit of using an array of const instead of const directly ?

Reply:
The benefit is limited to code readability. Grouping the 3 `const VERSION_XXX` consts under one array seemed more convenient and discoverable by developers, it also gives a stronger limit to the scope of values that `$version` can take.